src/map.cpp

```diff
@@ -7991,6 +8031,7 @@ void map::do_vehicle_caching( int z )
     auto &outside_cache = ch.outside_cache;
     auto &transparency_cache = ch.transparency_cache;
     auto &floor_cache = ch.floor_cache;
+    validate_autopilot_cache();
```

Comment:
Is this validation call really needed? right now it tanks performance even more than before :/

Ideally you wouldn't even need to have such a validation function if you catch every case where vehicles need to be add/removed from the cache.

Reply:
This would be the case where a submap is loaded back in , and vehicles that are loaded back in are added back to the cache.
I cant call g->m.add_to_autopilot_cache() for each vehicle, from savegame_json.cpp where submap::load() is called, as its not accessible there, but I didnt know where else to load them back into the cache when they enter reality bubble, any ideas?